lib/core/include/irods/irods_load_plugin.hpp

```diff
@@ -94,6 +101,274 @@ namespace irods
 
     } // resolve_plugin_path
 
+    /**
+     * \brief attempt to parse DT_FLAGS_1 to identify flags to pass to dlopen()
+     *
+     * \since 4.3.1
+     **/
```

Comment:
Let's stick with `///` since that appears to have become the convention.

Also, missing `\param[in] _plugin_path description ...`, `\throws`, and `\returns`.

Reply:
To my understanding, the only potential source of exceptions from this function would be the logging in the exception handlers, so I'm not sure what to put for `\throws`.

Reply:
The logger uses nlohmann::json, fmtlib, etc.

I think std::exception is good enough. At the very least, anyone calling this needs to catch std::exception to be safe.

If it helps, I've started leaving out **why** a particular exception may be thrown. I find that the message can quickly become incorrect over time and therefore it's safer to just let the dev know which exceptions can be thrown. They can interrogate the `.what()` for the reason.